### PR TITLE
Chore: Improve settings screen UI design

### PR DESCRIPTION
**Pull request by bikraj2**
Improved the Main Settings Screen:
Added three new screens:
AppDetails:Contains the app build ,commit, number, branch and Node Id
ShareLogs:Share logs
Force-Close Channel

### CODE REVIEW

**Comment by bonomat** [comment]
Thank you for your contribution @bikraj2 - this is greatly appreciated. 

The CI is failing on you because of formatting and (potential) lints. Can you run this please: 

- `just fmt`: this will format the code according to our standard
- `just lint-flutter`: this will run our linter and show you errors you will need to fix

File: mobile/ios/Podfile [hunk 1]
```diff
@@ -36,6 +36,17 @@ end
 
 post_install do |installer|
   installer.pods_project.targets.each do |target|
-    flutter_additional_ios_build_settings(target)
+      target.build_configurations.each do |config|
```
Comment (bonomat) [inline mobile/ios/Podfile:39]: Can you explain the reason for this change please?

File: mobile/ios/Podfile.lock [hunk 2]
```diff
@@ -149,6 +149,6 @@ SPEC CHECKSUMS:
   social_share: 702a5e3842addd22db515aa9e1e00a4b80a0296d
   url_launcher_ios: 08a3dfac5fb39e8759aeb0abbd5d9480f30fc8b4
 
-PODFILE CHECKSUM: cc1f88378b4bfcf93a6ce00d2c587857c6008d3b
+PODFILE CHECKSUM: 224e4680f2693f5d23f912e9aea395f4474ec34c
 
-COCOAPODS: 1.13.0
+COCOAPODS: 1.12.1
```
Comment (bonomat) [inline mobile/ios/Podfile.lock:154]: Please don't commit the changes in this file.

File: .DS_Store
Comment (bonomat) [inline .DS_Store:1]: Let's add these files to gitignore please: 

```
echo ".DS_Store" >> .gitignore
```

File: mobile/lib/common/settings_force_close.dart [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+                            letterSpacing: 0.4),
+                        children: [
+                          TextSpan(
+                            text: "Warning",
+                            style: TextStyle(
+                                color: Colors.red, fontWeight: FontWeight.w600),
+                          ),
+                          TextSpan(
+                            text:
+                                ": Force-closing your channel should only be considered as a last resort if 10101 is not reachable.\n\n\n",
```
Comment (bonomat) [inline mobile/lib/common/settings_force_close.dart:78]: Two new lines are enough :) 

```suggestion
                                ": Force-closing your channel should only be considered as a last resort if 10101 is not reachable.\n\n",
```

File: mobile/lib/common/settings_force_close.dart [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+                              text: "force-close",
+                              style: TextStyle(
+                                  color: Colors.red,
+                                  fontWeight: FontWeight.w600)),
+                          TextSpan(
+                              text:
+                                  ", you will have to pay the fees for going on-chain.\n\n"),
+                          TextSpan(
+                              text:
+                                  "Your funds can be claimed by your on-chain wallet after a X blocks.\n\n"),
```
Comment (bonomat) [inline mobile/lib/common/settings_force_close.dart:94]: We will update this with more details later. For now let's just leave it as this.

```suggestion
                                  "Your funds can be claimed by your on-chain after a while.\n\n"),
```

File: mobile/lib/common/settings_force_close.dart [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+                          TextSpan(
+                              text:
+                                  "You will have to open your app to claim your funds.\n")
+                        ],
```
Comment (bonomat) [inline mobile/lib/common/settings_force_close.dart:98]: I think we can skip it. 
```suggestion
```

File: mobile/lib/common/settings_details.dart [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+                                  info: widget.number,
+                                  lessInfo: true),
+                              moreInfo(context,
+                                  title: "Version",
+                                  info: widget.version,
+                                  lessInfo: true),
+                              moreInfo(context,
+                                  title: "Commit Hash", info: widget.branch),
+                              moreInfo(context,
+                                  title: "Branch", info: "Main", lessInfo: true)
```
Comment (bonomat) [inline mobile/lib/common/settings_details.dart:115]:
```suggestion
                              moreInfo(context,
                                  title: "Branch", info: info: widget.branch, lessInfo: true)
```

File: mobile/lib/common/settings_details.dart [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+                              moreInfo(context,
+                                  title: "Number",
+                                  info: widget.number,
+                                  lessInfo: true),
+                              moreInfo(context,
+                                  title: "Version",
+                                  info: widget.version,
+                                  lessInfo: true),
+                              moreInfo(context,
+                                  title: "Commit Hash", info: widget.branch),
```
Comment (bonomat) [inline mobile/lib/common/settings_details.dart:113]: This should be: 
```suggestion
                                  title: "Commit Hash", info: widget.commitHash),
```

File: mobile/lib/common/settings_details.dart [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+                ],
+              ),
+            ),
+          )),
+    );
+  }
+}
+
+Widget moreInfo(BuildContext context,
+    {required String title, required String info, bool lessInfo = false}) {
```
Comment (bonomat) [inline mobile/lib/common/settings_details.dart:129]: The variable `lessInfo` could be more expressive. It's not clear that it actually adds a copy & paste button.

File: mobile/lib/common/settings_details.dart [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+        ),
+        !lessInfo
+            ? GestureDetector(
+                onTap: () async {
+                  await Clipboard.setData(ClipboardData(text: info));
+                },
+                child: Icon(
+                  Icons.copy,
+                  size: 17,
+                  color: Colors.green.shade300,
```
Comment (bonomat) [inline mobile/lib/common/settings_details.dart:170]: Can you make this `tenTenOnePurple` please? Feel free to use a `shade...` to make it look good.

**Review by bonomat — COMMENTED** [review]
Good job. Looks good already. I've added a few comments. Feel free to ask if you have any questions.

File: mobile/ios/Podfile.lock (hunk 2 above)
Comment (bikraj2) [reply mobile/ios/Podfile.lock:154]: Will avoid this from next time.

File: mobile/ios/Podfile (hunk 1 above)
Comment (bikraj2) [reply mobile/ios/Podfile:39]: I had an issue as shown in this : https://stackoverflow.com/questions/77136958/flutter-or-react-native-xcode-15-error-xcode-dt-toolchain-dir-cannot-be-used.

**Comment by bikraj2** [comment]
@bonomat when I do just lint-flutter it shows no issues on my local device

**Comment by bonomat** [comment]
> @bonomat when I do just lint-flutter it shows no issues on my local device

You are right, `just lint-flutter` is ok, but you need to run `just fmt`.

File: mobile/ios/Podfile (hunk 1 above)
Comment (holzeis) [reply mobile/ios/Podfile:39]: Based on the accepted stackoverflow comment you referenced, that issue should have been resolved with the latest cocoapods version 

https://github.com/CocoaPods/CocoaPods/releases/tag/1.13.0

There was another change where the podfile.lock downgraded the cocoapods version to 1.12.1. I suspect that change is not need if you'd update your cocoapods to 1.13.0. Please do so and address both comments.

File: mobile/lib/common/settings_force_close.dart [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+class ForceClose extends StatefulWidget {
+  const ForceClose({
+    super.key,
+  });
+
+  @override
+  State<ForceClose> createState() => _ForceCloseState();
+}
+
+class _ForceCloseState extends State<ForceClose> {
```
Comment (holzeis) [inline mobile/lib/common/settings_force_close.dart:16]: 🔧 This state is not needed, please change to a stateless widget.

**Review by holzeis — COMMENTED** [review]
Great work! I left some minor remarks that would be great to get addressed before merging. But it already look very nice! Thank you!

**Comment by holzeis** [comment]
@bikraj2 I am taking this over and push it over the finish line. Thanks a lot for all the work that you have put into this! 👏

**Comment by bikraj2** [comment]
You are welcome. I would love to keep working on this project as a contributor. Been kinda of busy now so couldnot finish this.

**Pull request merged**